Dockerfile

```diff
@@ -30,8 +30,6 @@ WORKDIR /go/src/github.com/vmware-tanzu/velero
 
 COPY . /go/src/github.com/vmware-tanzu/velero
 
-RUN go mod download
```

Comment:
Why remove `go mod download` with this particular PR?

Reply:
changes to `Dockerfile` is unrelated to this PR. Removing this reduced my container build time by 30s.
Going to make a different PR with this change